### PR TITLE
Campaigns: Fix some matching issues (SH-330)

### PR DESCRIPTION
**Pull request by tulimaki**
* Avoid matching inactive filters and conditions
* Fix bug in product type catalog filter matching

### CODE REVIEW

File: shuup/campaigns/models/campaigns.py [hunk 1]
```diff
@@ -118,6 +118,12 @@ class CatalogCampaign(Campaign):
     def __str__(self):
         return force_text(_("Catalog Campaign: %(name)s" % dict(name=self.name)))
 
+    def save(self, *args, **kwargs):
+        super(CatalogCampaign, self).save(*args, **kwargs)
+        if not self.active:
+            self.filters.update(active=self.active)
+            self.conditions.update(active=self.active)
```
Comment (Pikkupomo) [inline shuup/campaigns/models/campaigns.py]: Should this work other way around too?

File: shuup/campaigns/models/campaigns.py [hunk 2]
```diff
@@ -204,6 +210,11 @@ class BasketCampaign(Campaign):
     def __str__(self):
         return force_text(_("Basket Campaign: %(name)s" % dict(name=self.name)))
 
+    def save(self, *args, **kwargs):
+        super(BasketCampaign, self).save(*args, **kwargs)
+        if not self.active:
+            self.conditions.update(active=self.active)
```
Comment (Pikkupomo) [inline shuup/campaigns/models/campaigns.py]: Should this work other way around too?

File: shuup/campaigns/models/campaigns.py (hunk 1 above)
Comment (tulimaki) [inline shuup/campaigns/models/campaigns.py]: Hmm yes maybe better to just always update the `active` to be same with the campaign.

**Comment by tulimaki** [comment]
Added the fix for the saves.

**Pull request merged**